File: client/src/app/_models/finance/voucher.ts
import { IVoucherAttachment } from "./voucherAttachment";
import { IVoucherEntry, VoucherEntry } from "./voucherEntry";

export interface IVoucher
{
	partyName: string;
	id: number;
	divn: string;
	coaId: number;
	accountName: string;
	voucherNo: number;
	voucherDated: Date;
	username: string;
	reviewedByUsername: string;
	reviewedOn: Date;
	amount: number;
	narration: string;
	/*approved: boolean;
	approvedByEmployeeId: number;
	approvedOn: Date;
	*/
	voucherEntries: IVoucherEntry[];
	voucherAttachments: IVoucherAttachment[];
}

export class Voucher implements IVoucher
{
	partyName='';
	id=0;
	divn = '';
	coaId = 0;
	accountName='';
	voucherNo = 0;
	voucherDated = new Date();
	username = '';
	reviewedByUsername = '';
	reviewedOn=new Date();
	amount = 0;
	narration = '';
	/*approved= false;
	approvedByEmployeeId=0;
	approvedOn=new Date();
	*/
	voucherEntries: VoucherEntry[]=[];
	voucherAttachments: IVoucherAttachment[]=[];
}

File: api/Entities/Finance/Voucher.cs
using System.ComponentModel.DataAnnotations;

namespace api.Entities.Finance
{
    public class VoucherTODELETE: BaseEntity
    {
        public VoucherTODELETE()
        {
        }

        //public string PartyName { get; set; }
        public int CVRefId { get; set; }
        [MaxLength(1)]
        public string Divn {get; set;}
        public int COAId { get; set; }
        public string AccountName { get; set; }
        [MaxLength(10), Required]
        public int VoucherNo { get; set; }
        public DateTime VoucherDated {get; set;}
        public long Amount {get; set;}
        public string Narration {get; set;}
        [Required]
        public ICollection<VoucherItem> VoucherItems { get; set; }
        public ICollection<VoucherAttachment> VoucherAttachments { get; set; }
        
    }
}